library/Notifications/Widget/ItemList/EscalationRecipientListItem.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/* Icinga Notifications Web | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Notifications\Widget\ItemList;
+
+use ipl\Html\BaseHtmlElement;
+use ipl\Html\FormElement\BaseFormElement;
+use ipl\Html\FormElement\SubmitButtonElement;
+
+class EscalationRecipientListItem extends BaseHtmlElement
+{
+    protected $defaultAttributes = ['class' => 'option'];
+
+    protected $tag = 'li';
+
+    /** @var ?SubmitButtonElement Remove button for the recipient */
+    public $removeButton;
+
+    /** @var BaseFormElement Recipient name field */
+    public $recipient;
+
+    /** @var BaseFormElement Recipient channel field */
+    public $channel;
+
+    public function __construct(
+        BaseFormElement $reipient,
+        BaseFormElement $channel,
+        ?SubmitButtonElement $removeButton
+    ) {
+        $this->recipient = $reipient;
+        $this->channel = $channel;
+        $this->removeButton = $removeButton;
+    }
+
+    protected function assemble(): void
+    {
+        $this->add([
+            $this->recipient,
+            $this->channel,
+            $this->removeButton
+        ]);
```

Comment:
You can call this method directly in constructor.

Reply:
There is a reason, this is done in assemble. We want the recipient list assembled as late as possible, as in case of single recipient the remove button would be remove from recipient fieldset.